pkg/client/tx/client.go

```diff
@@ -290,7 +292,15 @@ func (txnClient *txClient) SignAndBroadcast(
 		return either.SyncErr(err)
 	}
 
-	txResponse, err := txnClient.txCtx.BroadcastTx(txBz)
+	txResponse, err := retry.Call(func() (*cosmostypes.TxResponse, error) {
+		response, txErr := txnClient.txCtx.BroadcastTx(txBz)
+		// Wrap timeout height error to make it non-retryable.
```

Comment:
I'm wondering whether this may behave any differently/unexpectedly when using offline mode. :thinking: 

Perhaps not, but I just wanted to ask that you think about it as well for a moment and #PUC if you think of anything that's worth noting in this context.

Reply:
Can't think of any misbehavior w.r.t. offline mode.

I refactored the `SignAndBroadcast` to have a timeout height arg though. The caller knows best when the tx should land (e.g. `CreateClaim` should timeout when `ClaimWindowCloseHeight` is reached)